server.py

```diff
@@ -22,20 +22,29 @@
 
 app = Flask(__name__)
 
-class DoorOpener(object):
+class DoorManager(object):
     def __init__(self):
         self.open_door_ts = 0.0
         self.open_cond = threading.Condition()
+        self.locked = True
```

Comment:
Architectural request:
The DoorManager is becoming a more complex state machine now and the contracts of the various functions below intermingle and expose some internal information (eg the unlock function is called when the door is buzzed).

Can we write out the state machine explicitly in a comment here, and design the code to look more like the state machine? Eg naming the methods after the triggers and letting the DoorManager decide how to interpret the trigger. States seem to be something like

```
[LOCKED] -> [BUZZED] -> [SHOULD_OPEN] -> [NEUTRAL]
```
Triggers appear to be "BUZZ", "OPEN_TXT", and "LONGPOLL", and potentially 10 seconds passing triggers SHOULD_OPEN back to neutral too.

This diff appears to be adding a transition fro BUZZED back to LOCK with a new "LOCK_TXT" trigger

I think it will be more straightforward to interpret correctness with the state being the primary indicator of which fields are relevant.

Reply:
As a thought, it might be easier to interpret to do in two PR's
1) Switch to a state machine
2) Add the new state to the state machine

I think PR2 would be pretty short and sweet after 1